modules/gdscript/editor/gdscript_highlighter.cpp

```diff
@@ -465,7 +465,8 @@ Dictionary GDScriptSyntaxHighlighter::_get_line_syntax_highlighting_impl(int p_l
 
 		if (!in_node_ref && in_region == -1 && (str[j] == '$' || (str[j] == '%' && !is_binary_op))) {
 			in_node_ref = true;
-		} else if (in_region != -1 || (is_a_symbol && str[j] != '/' && str[j] != '%')) {
+		} else if (in_region != -1 || (is_a_symbol && str[j] != '/' && str[j] != '%') || (is_a_digit && j > 0 && (str[j - 1] == '$' || str[j - 1] == '/' || str[j - 1] == '%'))) {
```

Comment:
I'm not familiar with that code, but isn't it possible to rely on the same method that guarantees a valid identifier for nodes to do the completion, instead of reimplementing it?
We have `String::is_valid_identifier()` which I think handles this, though not 100% it's consistent with what is done here, or what `Node::validate_child_name()` does.

Reply:
Valid node names are not valid identifiers, however, only valid identifiers are allowed in node refs (I think it's to make parsing easier - since many of the allowed symbols have special meanings in GDScript.)

Either way, it'd be more inconvenient to do that, since this part of the highlighter works character by character to figure out if it should flip bools.

Reply:
Makes sense. You might be interested in checking if `core/string/char_utils.h` has useful functions to reuse in this parser, but this PR can be merged as is in the meantime.